src/test/unit/lib/mojaloop-requests/mojaloopRequests.test.js

```diff
@@ -0,0 +1,403 @@
+            },
+            jwsSign: true,
+            jwsSignPutParties: true,
+            jwsSigningKey: jwsSigningKey
+        };
+
+        let reqArgs;
+
+        let stub = sinon.stub(request, 'Request').callsFake((...args) => {
+            reqArgs = args;
+            return Promise.resolve({
+                statusCode: 200,
+                headers: {
+                    'content-length': 0
+                }
+            });
+        });
+
+        const testMr = new mr(conf);
+        const ret = await testMr.putParties('MSISDN', '123456', { fspid: 'dummy' }, 'dummy');
+
+        request.Request.restore();
+
+        t.assert(reqArgs[0].headers['fspiop-signature']);
+
+        t.pass();
+    } catch (err) {
+        t.fail(err.stack || util.inspect(err));
+    }
+});
+
+
+test.serial('does not sign put parties when jwsSign is true and jwsSignPutParties is false', async t => {
+    try {
+        const conf = {
+            logger: console,
+            tls: {
+                mutualTLS: {
+                    enabled: false
+                }
+            },
+            jwsSign: true,
+            jwsSignPutParties: false,
+            jwsSigningKey: jwsSigningKey
+        };
+
+        let reqArgs;
+
+        let stub = sinon.stub(request, 'Request').callsFake((...args) => {
+            reqArgs = args;
+            return Promise.resolve({
+                statusCode: 200,
+                headers: {
+                    'content-length': 0
+                }
+            });
+        });
+
+        const testMr = new mr(conf);
+        const ret = await testMr.putParties('MSISDN', '123456', { fspid: 'dummy' }, 'dummy');
+
+        request.Request.restore();
+
+        t.assert(typeof (reqArgs[0].headers['fspiop-signature']) === 'undefined');
+
+        t.pass();
+    } catch (err) {
+        t.fail(err.stack || util.inspect(err));
+    }
+});
+
+
+test.serial('does not sign put parties when jwsSign and jwsSignPutParties are false', async t => {
+    try {
+        const conf = {
+            logger: console,
+            tls: {
+                mutualTLS: {
+                    enabled: false
+                }
+            },
+            jwsSign: false,
+            jwsSignPutParties: false,
+            jwsSigningKey: jwsSigningKey
+        };
+
+        let reqArgs;
+
+        let stub = sinon.stub(request, 'Request').callsFake((...args) => {
+            reqArgs = args;
+            return Promise.resolve({
+                statusCode: 200,
+                headers: {
+                    'content-length': 0
+                }
+            });
+        });
+
+        const testMr = new mr(conf);
+        const ret = await testMr.putParties('MSISDN', '123456', { fspid: 'dummy' }, 'dummy');
+
+        request.Request.restore();
+
+        t.assert(typeof (reqArgs[0].headers['fspiop-signature']) === 'undefined');
+
+        t.pass();
+    } catch (err) {
+        t.fail(err.stack || util.inspect(err));
+    }
+});
```

Comment:
There are also signs of boilerplate code, so probably tests can be simplified like this:
1) Declare a function like this:
```
async function testPutParties(t, config, expectUndefined) {
  try {
       // Everything is false by default
        const conf = {
            logger: console,
            tls: {
                mutualTLS: {
                    enabled: false
                }
            },
            jwsSign: false,
            jwsSignPutParties: false,
            jwsSigningKey: jwsSigningKey,
           ...config
        };

       const stub = sinon.stub(request, 'Request');
       stub.callsFake(() => Promise.resolve({
          statusCode: 200,
          headers: {
             'content-length': 0
          },
       }));

        const testMr = new mr(conf);
        const ret = await testMr.putParties('MSISDN', '123456', { fspid: 'dummy' }, 'dummy');

        request.Request.restore();

        if (expectUndefined) {
           t.assert(typeof reqArgs[0].headers['fspiop-signature'] === 'undefined');
        } else {
           t.assert(reqArgs[0].headers['fspiop-signature']);
        }

        t.pass();
    } catch (err) {
        t.fail(err.stack || util.inspect(err));
    }
}
```
2) Use it in tests:
```
test.serial('does not sign put parties when jwsSign is false and jwsSignPutParties is true', async t =>
         testPutParties(t, {  jwsSignPutParties: true }, true)
);
```